File: cli/src/package/Package.cpp
#include "Package.h"

#include <utility>

namespace package {
    Package::Package(std::string_view name, std::string_view description, std::vector<PackageVersion> versions)
        : name{name}, description{description}, versions{std::move(versions)} {}

    const std::string &Package::getName() const {
        return name;
    }

    const std::string &Package::getDescription() const {
        return description;
    }

    const std::vector<PackageVersion> &Package::getVersions() const {
        return this->versions;
    }
}

File: web/views/package.php
<?php
$title = $params['package']->getName();
?>
<div class="center-content">
    <h2><?= $params['package']->getName() ?></h2>
    <h3 class="package-detail-description"><?= $params['package']->getDescription() ?></h3>
</div>
<hr/>

<h3>Versions</h3>
<?php if (empty($params['package']->getVersions())): ?>
    <p>This package doesn't have any version</p>
<?php else: ?>
    <ul>
        <?php foreach ($params['package']->getVersions() as $version): ?>
            <li><a href="<?= $version->getDownloadUrl() ?>"><?= $version->getIdentifier() ?> (<?= $version->getPublishedDate() ?>)</a></li>
        <?php endforeach ?>
    </ul>
<?php endif ?>
